### PR TITLE
Release chart 1.8.0

### LINKED ISSUE #3453: Bump to gnutls 3.6.15-r1 or later in docker build and base image
**Describe the bug**
There is a security vulnerability in the `gnutls` library that has been resolved in the 3.6.15-r1 release.  `gnutls` is used in the alpine base image as well as the [flux dockerfile](https://github.com/fluxcd/flux/blob/21debdeec3012371937348b12434543b196925aa/docker/Dockerfile.flux#L5).  The issue was reported to alpine [here](https://gitlab.alpinelinux.org/alpine/aports/-/issues/12543) and was resolved [before the 3.12.5 release](https://gitlab.alpinelinux.org/alpine/aports/-/commits/3.12-stable) was created.

**Expected behavior**

We should be using reducing security vulnerabilities as much as possible, including bumping to newer versions of packages and newer versions of base images when possible.  We should bump the apline image constraint in the dockerfile to 3.12.5 as well as the version constraint for `gnutls` to `>3.6.15-r1`.

**Additional context**

- Flux version: 1.22.0
- Kubernetes version:
- Git provider:
- Container registry provider:

### PR DESCRIPTION
**Pull request by kingdonb**
Related to #3455

Fixes #3453

### CODE REVIEW

**Comment by kingdonb** [comment]
I will rebase this to run the tests again after 1.22.1 image is published. The end-to-end chart tests will probably read as failing before that actually happens, at least that's what I'd think will happen.

File: chart/flux/values.yaml [hunk 1]
```diff
@@ -203,7 +203,7 @@ memcached:
   enabled: true
   hostnameOverride:
   repository: memcached
-  tag: 1.5.20
+  tag: 1.6.9-alpine
```
Comment (hiddeco) [inline chart/flux/values.yaml:206]: Having doubts about combining a MINOR version update of a dependency with a PATCH update of the chart itself.

**Comment by kingdonb** [comment]
I have squashed everything, and tested the alpine version of memcached image works OK, with fewer CVEs in scan results.

File: chart/flux/values.yaml (hunk 1 above)
Comment (kingdonb) [reply chart/flux/values.yaml:206]: If you think it makes sense, I will amend the chart bump and call it 1.8.0

Comment (kingdonb) [reply chart/flux/values.yaml:206]: (Done)

**Review by hiddeco — APPROVED** [review]
Thank you @kingdonb :rainbow:

**Pull request merged**